### PR TITLE
Updated the docs

### LINKED ISSUE #58: Improve documentation
* [x] Add detailed rewrites documentation.
* [x] Document CLI flags.

### PR DESCRIPTION
**Pull request by grzkv**
Fixes #58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,15 +59,36 @@ To test the second store (alone or in conjunction) change the metric path to `te
...
 
-Please refer to the sample configs in the _config_ folder for examples and documentation.
+Please refer to the [sample config](config/config.toml) for examples and documentation.
+
+Routing
+-------
+
+Is defined in the [rules config](config/rules.toml) that is in turn referred to in the [main config](config/config.toml). This is how it works:
+- routing rules are applied to each incoming record in order;
+- `continue` variable defines whether to continue matching next rules. It is `true` by default;
```
Comment (azhiltsov) [inline README.md:77]: is it really defaults to true?
In our production config it used differently with assumption that it defaults to 'false'

File: README.md [hunk 2]
```diff
@@ -45,15 +59,36 @@ To test the second store (alone or in conjunction) change the metric path to `te
...
+- when regex in a rule matches, the record is sent to the clusters in the `clusters` list;
```
Comment (azhiltsov) [inline README.md:78]: If my previous comment right then I would rephrase this:
- First match win and matched record is sent to the clusters in the 'clusters' list;
- 'continue' variable defines whether to continue matching next rules and possible send the same metric to other 'clusters'. Default is 'false';

File: README.md [hunk 3]
```diff
@@ -45,15 +59,36 @@ To test the second store (alone or in conjunction) change the metric path to `te
...
+
+#### Rewrites
+
+Optionally, it is possible to apply the [rewrite rules](config/rewrite.toml). This is how they work:
+
+- all rules are applied to each record;
+- rewrites are applied before the routing;
+- rule matches if `from` matches;
+    - then metric path is rewriten to `to`;
+    - if `copy` is `true` the original metric is retained in addition to the modified one.
```
Comment (azhiltsov) [inline README.md:91]: Please clarify that rewritten results are passed directly to the routing 'rules' bypassing rewrites.
Unclear: can multiple 'from' match create multiple 'to' metrics? Does copy = 'true'/'false' affect the answer?

File: config/rewrite.toml [hunk 4]
```diff
@@ -1,10 +1,5 @@
-# Rewrite rules.
-# - all rules are applied to each record
-# - if rule mathches, metric path is rewriten
-# - if copy is true, new metric is created
-
 [[rewrite]]
```
Comment (azhiltsov) [inline config/rewrite.toml:1]: Please comment this block as this is default config.
We don't want to ship default config with rewrites.

Please add the next example which demonstrating how to use regex capturing group
[[rewrite]]
    from = '^sys\.([A-Za-z0-9_-]+)\.(.*)'
    to = 'sys.$2.$1'
    copy = true

File: README.md (hunk 2 above)
Comment (grzkv) [reply README.md]: Updated.

File: README.md (hunk 1 above)
Comment (grzkv) [reply README.md]: That's right. Updated.

For findings like this one is why we write the docs.

File: config/rewrite.toml (hunk 4 above)
Comment (grzkv) [reply config/rewrite.toml:1]: Default config has rewrites unset.

I may add this in the next iteration of the documentation. I would keep it simple for now.

File: README.md (hunk 3 above)
Comment (grzkv) [reply README.md:91]: This is why I think we should not have re-writes. They are not usable and opaque. They also create a lot of "magic": I sent A, but got B, but why? I hope we will remove them in the future.

I will add more explanations.
